SalesforceCommerceCloud/pwa-kit
# [Localization] Locale selector: update list of commonly-used locales

# Description

Ticket: https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000008vKWsYAM/view

For a better developer experience, whenever they add a new locale (e.g. by updating the 'pwa/package.json' file), they should not have to worry about configuring the locale selector UI too, if possible. This PR adds a list of commonly-used locale names and throws an error if the new locale is not found in this list.

## Performance Consideration

There is very little performance impact:
- The bundle size grows by 2KB minified (0.7 KB gzipped)
- And a similar growth in the server-generated html too 

# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- Have a list of commonly-used locales (for how this list is determined, see the conversation [here](https://gus.lightning.force.com/lightning/r/0D5EE00000AoXsf0AF/view))
- Throws an error when a developer adds a locale that isn't in the list of common locales
- Some refactoring for clarity

# How to Test-Drive This PR

1. Remove this `en-GB` from the list of common locales: https://github.com/SalesforceCommerceCloud/pwa-kit/blob/243a364978278c4ab7061ae08fc3b229a8676e68/packages/pwa/app/constants.js#L72
2. `npm run build-translations`
3. `npm start`
4. Load homepage http://localhost:3000/
5. You should see a server error page that shows a unique error message

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
